tests/utils.go

```diff
@@ -0,0 +1,106 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package tests
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"fmt"
+	"math/big"
+	"strconv"
+	"strings"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/subnet-evm/core/types"
+	"github.com/ava-labs/subnet-evm/ethclient"
+	"github.com/ava-labs/subnet-evm/params"
+	"github.com/ava-labs/subnet-evm/tests/utils"
+	"github.com/ava-labs/subnet-evm/tests/utils/runner"
+	"github.com/ethereum/go-ethereum/common"
+	. "github.com/onsi/gomega"
+)
+
+var (
+	defaultTeleporterMessageGas       uint64 = 200_000
+	defaultTeleporterMessageGasFeeCap        = big.NewInt(225 * params.GWei)
+	defaultTeleporterMessageGasTipCap        = big.NewInt(params.GWei)
+	defaultTeleporterMessageValue            = common.Big0
+)
+
+// Teleporter contract sendCrossChainMessage input type
+type TeleporterMessageInput struct {
+	DestinationChainID      ids.ID
+	DestinationAddress      common.Address
+	FeeInfo                 FeeInfo
+	RequiredGasLimit        *big.Int
+	Message                 []byte
+	AllowedRelayerAddresses []common.Address
+}
+
+type FeeInfo struct {
+	ContractAddress common.Address
+	Amount          *big.Int
+}
+
+func httpToWebsocketURI(uri string, blockchainID string) string {
+	return fmt.Sprintf("ws://%s/ext/bc/%s/ws", strings.TrimPrefix(uri, "http://"), blockchainID)
+}
+
+func httpToRPCURI(uri string, blockchainID string) string {
+	return fmt.Sprintf("http://%s/ext/bc/%s/rpc", strings.TrimPrefix(uri, "http://"), blockchainID)
+}
+
+func getURIHostAndPort(uri string) (string, uint32, error) {
+	// At a minimum uri should have http:// of 7 characters
+	Expect(len(uri)).Should(BeNumerically(">", 7))
+	if uri[:7] == "http://" {
+		uri = uri[7:]
+	} else if uri[:8] == "https://" {
+		uri = uri[8:]
+	} else {
+		return "", 0, fmt.Errorf("invalid uri: %s", uri)
+	}
+
+	// Split the uri into host and port
+	hostAndPort := strings.Split(uri, ":")
+	Expect(len(hostAndPort)).Should(Equal(2))
+
+	// Parse the port
+	port, err := strconv.ParseUint(hostAndPort[1], 10, 32)
+	if err != nil {
+		return "", 0, fmt.Errorf("failed to parse port: %w", err)
+	}
+
+	return hostAndPort[0], uint32(port), nil
+}
+
+func newTestTeleporterMessage(chainIDInt *big.Int, teleporterAddress common.Address, nonce uint64, data []byte) *types.Transaction {
```

Comment:
nit: can we rename to `newTestTeleporterTransaction` and replace Message with Transaction in the default values